src/packages/solid/components/Keyboard/Keyboard.tsx

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2023 Comcast Cable Communications Management, LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { type Component, createSignal } from 'solid-js';
+import { type IntrinsicNodeProps } from '@lightningjs/solid';
+import styles, { type KeyboardFormat } from './Keyboard.styles.js';
+import KeyboardSimple from './KeyboardSimple.jsx';
+import type { KeyProps } from '../Key/Key.jsx';
+
+export interface KeyboardProps extends IntrinsicNodeProps {
+  /**
+   * object containing arrays that represent different formats that the keyboard can be presented in. These arrays can contain strings or objects.
+   */
+  formats: KeyboardFormat;
+
+  /**
+   * center the keyboard within it's set width (must set the w property of Keyboard)
+   */
+  centerKeyboard?: boolean;
+
+  /**
+   * wrap keyboard rows when navigating left and right
+   */
+  rowWrap?: boolean;
```

Comment:
Im so curious what use case this had, when wouldn't we want to wrap the nav?

Reply:
if a user get to the end of the keyboard and wants it to wrap back to the front instead of having to continuously click the left key

Reply:
i get that, I'm questioning why _not_ wrapping ever considered a use case

Reply:
keep this for now, im curious to look into focus handling and if this'll need to be implemented differently in solid